x-pack/plugins/reporting/server/test_helpers/create_mock_reportingplugin.ts

```diff
@@ -57,12 +59,58 @@ const createMockPluginStart = (
   };
 };
 
-export const createMockConfigSchema = (overrides?: any) => ({
-  index: '.reporting',
-  kibanaServer: { hostname: 'localhost', port: '80' },
-  capture: { browser: { chromium: { disableSandbox: true } } },
-  ...overrides,
-});
+interface ReportingConfigTestType {
+  index: string;
+  encryptionKey: string;
+  queue: Partial<ReportingConfigType['queue']>;
+  kibanaServer: Partial<ReportingConfigType['kibanaServer']>;
+  csv: Partial<ReportingConfigType['csv']>;
+  capture: any;
+  server?: any;
+}
+
+export const createMockConfigSchema = (
+  overrides: Partial<ReportingConfigTestType> = {}
+): ReportingConfigTestType => {
+  // deeply merge the defaults and the provided partial schema
+  return {
+    index: '.reporting',
+    encryptionKey: 'cool-encryption-key-where-did-you-find-it',
+    ...overrides,
+    kibanaServer: {
+      hostname: 'localhost',
+      port: 80,
+      ...overrides.kibanaServer,
+    },
+    capture: {
+      browser: {
+        chromium: {
+          disableSandbox: true,
+        },
+      },
+      ...overrides.capture,
+    },
+    queue: {
+      timeout: 120000,
+      ...overrides.queue,
+    },
+    csv: {
+      ...overrides.csv,
+    },
+  };
+};
```

Comment:
All of these config properties should be defined, because we're now calling conversion function on the value if it is not a scalar type. Doing that on an undefined (mock) value could throw an error.